src/legacy/ui/public/courier/search_source/search_source.js

```diff
@@ -570,7 +571,9 @@ export function SearchSourceProvider(Promise, Private, config) {
           const fields = flatData.fields;
           if (fields) {
             // filter out the docvalue_fields, and script_fields to only include those that we are concerned with
-            flatData.body.docvalue_fields = _.intersection(flatData.body.docvalue_fields, fields);
+            flatData.body.docvalue_fields = flatData.body.docvalue_fields.filter(docValue => {
+              return fields.includes(docValue.field);
+            });
```

Comment:
Do we know for a fact that docValues are always an object now? Couldn't a consumer of SearchSource pass in a string? I assume they were only strings at one point in time, and the ES API still accepts strings. Might be good to have a conditional here to handle both cases.

Reply:
Looks like docvalue_fields was an array of strings upto 6.3, https://www.elastic.co/guide/en/elasticsearch/reference/6.3/search-request-docvalue-fields.html. And in Kibana, getComputedFields set docvalue_fields as an array of strings https://github.com/elastic/kibana/blob/6.2/src/ui/public/index_patterns/_get_computed_fields.js#L9.

Here is the PR that changed docvalue_fields to an array of objects https://github.com/elastic/kibana/pull/22771. It targets 6.4.1.

From that PRs description, it looks like using array of strings was deprecated in 7.0. Do you really want logic in place for ES requests that are invalid and will fail?

Reply:
The PR you linked is only talking about date fields. According to the ES docs, the `docvalue_fields` param can be an array of both strings and objects, even in master.

Reply:
Thanks, I missed that. I have updated the logic to handle strings and objects